store/rootmulti/store.go

```diff
@@ -576,6 +576,52 @@ func (rs *Store) pruneStores() error {
 	return nil
 }
 
+// PruneHistoryVersions prunes the history versions of each substore based on
+// the pruning options.
+func (rs *Store) PruneHistoryVersions() error {
+	pruningOptions := rs.GetPruning()
+	latestVersion := getLatestVersion(rs.db)
+	pruningVersions := []int64{}
+
+	if pruningOptions.Interval == 0 {
+		rs.logger.Info("pruning returns with no op because pruning interval is set to 0\n")
+		return nil
+	}
+
+	for i := int64(0); i < latestVersion; i++ {
+		if (i%int64(pruningOptions.Interval) != 0) && (i+int64(pruningOptions.KeepRecent) < latestVersion) {
+			pruningVersions = append(pruningVersions, i)
+		}
+	}
+
+	if len(pruningVersions) == 0 {
+		rs.logger.Info("pruning returns with no op because no matched versions to prune\n")
+		return nil
+	}
+
+	for key, store := range rs.stores {
+		// If the store is wrapped with an inter-block cache, we must first unwrap
+		// it to get the underlying IAVL store.
+		if store.GetStoreType() != types.StoreTypeIAVL {
+			rs.logger.Info("can not prune a non-IAVL store for store with key: %v\n", key)
+			continue
+		}
+
+		store = rs.GetCommitKVStore(key)
+
+		err := store.(*iavl.Store).DeleteVersions(pruningVersions...)
```

Comment:
It would be more efficient if we can preprocess it into ranges and call `DeleteVersionsRange`.